### ISSUE TITLE
[Feat] Add fine-tune training API

### ISSUE THREAD

**Issue by YuanmingLeee** [cap-ntu/ML-Model-CI#198]
#### Problem description

This is a feature request: Add fine-tune training APIs, so that users can submit fine-tune jobs to the system.

#### Steps to Reproduce the Problem
N.A.

#### Expected Behavior
* User can submit a fine-tune job with the specified dataset, optimizer, learning rate...
* User can check the fine-tune job status
* User can terminate and delete the on-going fine-tune job
